src/mamba/py_interface.cpp

```diff
@@ -212,19 +215,33 @@ PYBIND11_MODULE(mamba_api, m)
         .def(py::init<const std::string&>())
         .def("load", &PrefixData::load);
 
-    py::class_<Channel>(m, "Channel")
-        .def(py::init([](const std::string& value) { return &(make_channel(value)); }))
+    py::class_<Channel, std::unique_ptr<Channel, py::nodelete>>(m, "Channel")
+        .def(py::init([](const std::string& value)
+                      { return const_cast<Channel*>(&make_channel(value)); }))
         .def_property_readonly("scheme", &Channel::scheme)
         .def_property_readonly("location", &Channel::location)
         .def_property_readonly("name", &Channel::name)
-        .def_property_readonly("platform", &Channel::platform)
-        .def_property_readonly("subdir", &Channel::platform)
+        .def_property_readonly("platforms", &Channel::platforms)
         .def_property_readonly("canonical_name", &Channel::canonical_name)
-        .def("url", &Channel::url, py::arg("with_credentials") = true)
-        .def("__repr__", [](const Channel& c) { return join_url(c.name(), c.platform()); });
+        .def("urls", &Channel::urls, py::arg("with_credentials") = true)
+        .def("__repr__",
+             [](const Channel& c)
+             {
+                 auto s = c.name();
+                 s += "[";
+                 bool first = true;
+                 for (const auto& platform : c.platforms())
+                 {
+                     if (!first)
+                         s += ",";
+                     s += platform;
+                     first = false;
+                 }
+                 s += "]";
+                 return s;
+             });
 
     m.def("get_channel_urls", &get_channel_urls);
```

Comment:
It's not clear whether `get_channel_urls` is important to keep around or not. I didn't yet remove it because it is the only function in the C++ code which uses the `get_custom_multichannels` function, which seemed odd to me.